integration-tests/load-tests/signatureRegistry.test.ts

```diff
@@ -0,0 +1,160 @@
+    for (let i = 0; i < keysToTest.length; i++) {
+        const key = keysToTest[i];
+        let msaOption = await getMsaFromKey(key);
+        if (!msaOption.isSome) {
+            console.log(`The ${startingNumber + i} key (${key.address}) failed to be associated with an MSA. Trying another block...`);
+            await createBlock();
+            msaOption = await getMsaFromKey(key);
+
+        }
+        assert(msaOption.isSome, `The ${startingNumber + i} key (${key.address}) failed to be associated with an MSA.`);
+    }
+}
+
+// Generate MSAs and give it 100 UNIT
+async function generateMsas(count: number = 1): Promise<GeneratedMsa[]> {
+    // Make sure we are not on an edge
+    const createBlockEvery = count === 300 ? 290 : 300;
+
+    // Create and fund the control keys
+    let controlKeyPromises: Array<Promise<KeyringPair>> = [];
+    const devAccount = devAccounts[0].keys;
+    let devAccountNonce = await getNonce(devAccount);
+    for (let i = 0; i < count; i++) {
+        controlKeyPromises.push(createAndFundKeypair(100n * 10n * EXISTENTIAL_DEPOSIT, undefined, devAccount, devAccountNonce++));
+        if (i > 0 && i % createBlockEvery === 0) await createBlock(100);
+    }
+    await createBlock();
+    const controlKeys = await Promise.all(controlKeyPromises);
+
+    // Create the msas
+    let msaPromises: Array<Promise<GeneratedMsa>> = [];
+    for (let i = 0; i < count; i++) {
+        msaPromises.push(createMsa(controlKeys[i]).then((id) => ({
+                controlKey: controlKeys[i],
+                id,
+                nonce: 1,
+            }))
+        );
+        if (i > 0 && i % createBlockEvery === 0) {
+            await createBlock(200);
+            console.log("Generated Msas: ", i);
+        }
+    }
+    await createBlock(750);
+    // Create a second block in case there were more than could fit in ^ blocks
+    await createBlock();
+    const msaIds = await Promise.all(msaPromises)
+    console.log("Generated Msas: ", count);
+    return msaIds;
+}
+
+async function createBlock(wait: number = 300) {
+    // Wait ms before creating the block to give the chain time to process the transaction pool
+    await new Promise(r => setTimeout(r, wait));
+    return firstValueFrom(ExtrinsicHelper.api.rpc.engine.createBlock(true, true));
+}
+
+function getMsaFromKey(keys: KeyringPair): Promise<Option<u64>> {
+    return ExtrinsicHelper.apiPromise.query.msa.publicKeyToMsaId(keys.address);
+}
+
+async function getNonce(keys: KeyringPair): Promise<number> {
+    const nonce = await firstValueFrom(ExtrinsicHelper.api.call.accountNonceApi.accountNonce(keys.address));
+    return nonce.toNumber();
+}
+
+async function createMsa(keys: KeyringPair): Promise<u64> {
+    const result = await ExtrinsicHelper.createMsa(keys).signAndSend();
+    const msaRecord = result[1]["msa.MsaCreated"];
+    if (msaRecord) return msaRecord.data[0] as u64;
+
+    // This doesn't always work I think due to the load. So doing a backup call
```

Comment:
Overloading the WebSocket limit specifically. So those createMsa subscriptions are getting closed before they can respond.